test/e2e_new/broker_trigger_sink_test.go

```diff
@@ -0,0 +1,157 @@
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e_new
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cloudevents/sdk-go/v2/test"
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/kafka"
+	brokerreconciler "knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/broker"
+	testpkg "knative.dev/eventing-kafka-broker/test/pkg"
+	brokerconfigmap "knative.dev/eventing-kafka-broker/test/rekt/resources/configmap/broker"
+	"knative.dev/eventing-kafka-broker/test/rekt/resources/kafkasink"
+	"knative.dev/eventing-kafka-broker/test/rekt/resources/kafkatopic"
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	"knative.dev/eventing/test/rekt/resources/broker"
+	"knative.dev/eventing/test/rekt/resources/trigger"
+	"knative.dev/pkg/system"
+	"knative.dev/reconciler-test/pkg/eventshub"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/k8s"
+	"knative.dev/reconciler-test/pkg/knative"
+	"knative.dev/reconciler-test/pkg/resources/job"
+)
+
+const (
+	kafkaConsumerImage = "ko://knative.dev/eventing-kafka-broker/test/test_images/kafka-consumer"
+)
+
+/*
++---------+     +--------+     +---------+   +---------------------------+
+|  Broker +---->+ Trigger+---->+KafkaSink+-->+kafka consumer (test image)|
++----+----+     +--------+     +----+----+   +---------------------------+
+
+	|                              ^
+	|          +--------+          |
+	+--------->+ Trigger+----------+
+	           +--------+
+*/
+func TestBrokerTriggersSink(t *testing.T) {
+
+	// Run Test In Parallel With Others
+	t.Parallel()
+
+	//Create The Test Context / Environment
+	ctx, env := global.Environment(
+		knative.WithKnativeNamespace(system.Namespace()),
+		knative.WithLoggingConfig,
+		knative.WithTracingConfig,
+		k8s.WithEventListener,
+	)
+
+	brokerName := feature.MakeRandomK8sName("broker")
+	brokerConfig := feature.MakeRandomK8sName("brokercfg")
+	trigger1Name := feature.MakeRandomK8sName("trigger-1")
+	trigger2Name := feature.MakeRandomK8sName("trigger-2")
+	sink := feature.MakeRandomK8sName("sink")
+	verifyMessagesJobName := feature.MakeRandomK8sName("verify-messages-job")
+
+	topic := kafka.BrokerTopic(brokerreconciler.TopicPrefix, &eventingv1.Broker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      brokerName,
+			Namespace: env.Namespace(),
+		},
+	})
+
```

Comment:
We usually separate feature and test because it can useful to reuse them, even if in this case it doesn't seem useful but I'd do that for consistency.

Reply:
Yeah I didn't see a reason for reusing it too. Anyhow I updated it, so we have consistency :+1: